WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Blast.Core.Interfaces;
+using Blast.Core.Objects;
+using Blast.Core.Results;
+using System.Net.Http;
+using System.Drawing;
+using System.IO;
+using System.Net.Http.Json;
+using static WikiPreviewConsole.WikiResult;
+using System.Threading.Channels;
+using Dasync.Collections;
+using static WikiPreview.Fluent.Plugin.WikiPreviewSearchOperation;
+using System;
+using Blast.API.Core.Processes;
+using Blast.API.Processes;
+using System.Text.Json;
+using Blast.API.Search;
+
+namespace WikiPreview.Fluent.Plugin
+{
+    class WikiPreviewSearchApp : ISearchApplication
+    {
+        public static readonly string SearchAppName = "WikiPreview";
+        public static readonly string WikiSearchTagName = "Wiki";
+        private readonly string wikiRootUrl = "https://en.wikipedia.org/wiki/";
+        private readonly SearchApplicationInfo _applicationInfo;
+
+        public WikiPreviewSearchApp()
+        {
+
+            _applicationInfo = new SearchApplicationInfo(SearchAppName,
+                "This extension can search in Wikipedia", WikiPreviewSearchResult.supportedOperations)
+            {
+                MinimumSearchLength = 2,
+                IsProcessSearchEnabled = false,
+                IsProcessSearchOffline = false,
+                ApplicationIconGlyph = "\uE946",
+                SearchAllTime = ApplicationSearchTime.Fast,
+                DefaultSearchTags = WikiPreviewSearchResult.searchTags,
+            };
+        }
+
+
+        public SearchApplicationInfo GetApplicationInfo()
+        {
+            return _applicationInfo;
+        }
+
+        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        {
+            return new();
+        }
+
+        public ValueTask<IHandleResult> HandleSearchResult(ISearchResult searchResult)
+        {
+            ISearchOperation selectedOperation = searchResult.SelectedOperation;
+            Type selectedOperationType = selectedOperation.GetType();
+            IProcessManager managerInstance = ProcessUtils.GetManagerInstance();
+
+            switch (selectedOperationType)
+            {
+                case var type when type == typeof(WikiSearchOperation):
+                    {
+                        string wikiUrl = wikiRootUrl + searchResult.DisplayedName;
+                        managerInstance.StartNewProcess(wikiUrl);
+                        return new ValueTask<IHandleResult>(new HandleResult(true, false));
```

Comment:
No need to return from the `switch` if they all return the same value. Just move it out of the `switch`